client/src/components/pages/RobotInteractionCockpit/RobotInteractionCockpit.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import { Layout, Card, Steps, Space, Button, Typography } from 'antd';
+import HeaderNavbar from '../../content/HeaderNavbar/HeaderNavbar';
+import RobotInteractionInputSection from '../../content/RobotInteractionSections/RobotInteractionInputSection';
+import { getAllParametersForRobot } from '../../../api/variableRetrieval';
+import socket from '../../../utils/socket/socketConnections';
+import { upsert } from '../../../utils/attributeAndParamUtils';
+import styles from './RobotInteractionCockpit.module.css';
+import { isRobotExecutable } from '../../../utils/robotExecution';
+
+const { Step } = Steps;
+const { Title } = Typography;
+
+/**
+ * @description Page, where you can interact with a robot and for example enter input
+ * @category Client
+ * @component
+ */
+const RobotInteractionCockpit = (match) => {
+  const { robotId } = match.match.params;
+  const { userId } = match.location.state;
+  const isMounted = useRef(true);
+  const [parameterList, setParameterList] = useState([]);
+  const [currentStep, setCurrentStep] = useState(0);
+
+  /**
+   * @description For each activity of the current robot get the id, the name and all the parameters that require a user input
+   */
+  const getActivityAndParameterInformation = useCallback(() => {
+    getAllParametersForRobot(robotId)
+      .then((response) => response.json())
+      .then((parameterObjects) => {
+        const activityInformationList = [];
+        Array.prototype.forEach.call(parameterObjects, (parameterObject) => {
+          const { activityId } = parameterObject;
+          let activityName = '';
+          let ssot = sessionStorage.getItem('ssotLocal');
+          ssot = JSON.parse(ssot);
+          Array.prototype.forEach.call(ssot.elements, (elem) => {
```

Comment:
Exactly why is this needed as prototype? Shouldn't by the async nature of it ssot be resolved at this point?

Reply:
I am not quite sure if I understand what you mean. I want to iterate here over ssot.elements by using the .foreach function. The problem is that the type of ssot.elements (although it is an array) is Object and .foreach would not work. Thats why I use the Array.prototype.foreach construct. [Here](https://stackoverflow.com/questions/43743560/foreach-vs-array-prototype-foreach-call) is a stackoverflow thread (although not 100% our case here) it explains the use case for Array.prototype.foreach quite well